anchore_engine/services/apiext/api/controllers/utils.py

```diff
@@ -225,3 +229,245 @@ def validate_image_add_source(analysis_request_dict, api_schema):
 
     else:
         raise ValueError('Expected a "source" property in the input dict')
+
+
+def make_cvss_scores(metrics):
+    """
+     [
+        {
+          "cvss_v2": {
+            "base_metrics": {
+              ...
+            },
+            "vector_string": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
+            "version": "2.0"
+          },
+          "cvss_v3": {
+            "base_metrics": {
+             ...
+            },
+            "vector_string": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
+            "version": "3.0"
+          },
+          "id": "CVE-2019-1234"
+        },
+        {
+          "cvss_v2": {
+            "base_metrics": {
+              ...
+            },
+            "vector_string": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
+            "version": "2.0"
+          },
+          "cvss_v3": {
+            "base_metrics": {
+             ...
+            },
+            "vector_string": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
+            "version": "3.0"
+          },
+          "id": "CVE-2019-3134"
+        },
+     ]
+    :param metrics:
+    :return:
+    """
+    score_list = []
+
+    for metric in metrics:
+        new_score_packet = {
+            "id": metric.get("id"),
+        }
+        score_list.append(new_score_packet)
+
+        for i in [3, 2]:
+            cvss_dict = metric.get("cvss_v{}".format(i), {})
+            base_metrics = cvss_dict.get("base_metrics", {}) if cvss_dict else {}
+
+            tmp = base_metrics.get("base_score", -1.0)
+            base_score = float(tmp) if tmp else -1.0
+            tmp = base_metrics.get("exploitability_score", -1.0)
+            exploitability_score = float(tmp) if tmp else -1.0
+            tmp = base_metrics.get("impact_score", -1.0)
+            impact_score = float(tmp) if tmp else -1.0
+
+            new_score_packet["cvss_v{}".format(i)] = {
+                "base_score": base_score,
+                "exploitability_score": exploitability_score,
+                "impact_score": impact_score,
+            }
+
+    return score_list
+
+
+def make_response_vulnerability(vulnerability_type, vulnerability_data):
+    ret = []
+
+    if not vulnerability_data:
+        logger.warn("empty query data given to format - returning empty result")
+        return ret
+
+    eltemplate = {
```

Comment:
What does "el" stand for throughout here?

Reply:
this is copy-pasted code, el = element